.travis-build.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -e # fail on errors
+#set -x # echo commands run
+
+export toolset_branch=$1
+export server_type=$2
+export tags_list=$3
+
+
+export ubuntu_version=${ubuntu_version_name:-"trusty"}
+export ros_release=${ros_release_name:-"indigo"}
+export docker_image=${docker_image_name:-"shadowrobot/build-tools:$ubuntu_version-$ros_release"}
+
+export docker_user=${docker_user_name:-"user"}
+export docker_user_home=${docker_user_home_dir:-"/home/user"}
+
+export extra_variables="codecov_secure=$CODECOV_TOKEN github_login=$GITHUB_LOGIN github_password=$GITHUB_PASSWORD ros_release=$ros_release ubuntu_version_name=$ubuntu_version "
```

Comment:
I can understand your motivation of putting everything into `.travis.yml`. OTOH, sometimes it's nice to have a 'do everything' script.

If you put everything into `.travis.yml`, you might find YAML's multiline string literals with `>` and `|` handy.